### PR TITLE
Add duckplyr blog post

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

File: _posts/2024-04-02-duckplyr.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+## Conclusion
+
+The duckplyr package for R wraps DuckDB's state-of-the-art analytical query processing techniques in a dplyr-compatible API. We have gone to great lenghts to ensure compatibility despite switching execution paradigms from eager to lazy and having to translate expressions to a different environment. We continue to work to expand duckplyr's capabilites but would love to hear your experiences trying it out.
```
Comment (Mytherin) [inline _posts/2024-04-02-duckplyr.md:224]:
```suggestion
The duckplyr package for R wraps DuckDB's state-of-the-art analytical query processing techniques in a dplyr-compatible API. We have gone to great lengths to ensure compatibility despite switching execution paradigms from eager to lazy and having to translate expressions to a different environment. We continue to work to expand duckplyr's capabilites but would love to hear your experiences trying it out. 
```

File: _posts/2024-04-02-duckplyr.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+Here is an example of the duality of the result of duckplyr operations using R’s `inspect()` method:
+
+```R
+dd <- as_duckplyr_df(data.frame(n=1:10)) |> mutate(m=n+1)
+.Internal(inspect(dd))
+```
+
+```
```
Comment (Mytherin) [inline _posts/2024-04-02-duckplyr.md:169]:
```suggestion
```text
```

File: _posts/2024-04-02-duckplyr.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+2. somehow integrate existing state-of-the-art technology into interactive data analysis environments.
+
+The main issue with approach one is that building a *competitive* analytical query engine from scratch is a multi-million dollar effort requiring a team of highly specialized experts on query engine construction. There are many moving highly complex parts that all have to play together nicely. There are seemingly-obvious questions in query engines that one can [get a PhD in data management systems for a solution](https://hannes.muehleisen.org/publications/ICDE2023-sorting.pdf). Recouping such a massive investment in a space where it is common that tools are built by volunteers in their spare time and released for free is challenging. That being said, there are a few commendable projects in this space like [data.table]([https://CRAN.R-project.org/package=data.table](https://cran.r-project.org/package=data.table)) or more recently [pola.rs](https://pola.rs) that - while they offer greatly improved performance over older tools - still fall short from the state of the art in analytical data management systems in many areas.
```
Comment (Mytherin) [inline _posts/2024-04-02-duckplyr.md:36]:
```suggestion
The main issue with approach one is that building a *competitive* analytical query engine from scratch is a multi-million dollar effort requiring a team of highly specialized experts on query engine construction. There are many moving highly complex parts that all have to play together nicely. There are seemingly-obvious questions in query engines that one can [get a PhD in data management systems for a solution](https://hannes.muehleisen.org/publications/ICDE2023-sorting.pdf). Recouping such a massive investment in a space where it is common that tools are built by volunteers in their spare time and released for free is challenging. That being said, there are a few commendable projects in this space like [data.table]([https://CRAN.R-project.org/package=data.table](https://cran.r-project.org/package=data.table)) or more recently [pola.rs](https://pola.rs) that offer greatly improved performance over older tools.
```

File: _posts/2024-04-02-duckplyr.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+## Conclusion
+
+The duckplyr package for R wraps DuckDB's state-of-the-art analytical query processing techniques in a dplyr-compatible API. We have gone to great lenghts to ensure compatibility despite switching execution paradigms from eager to lazy and having to translate expressions to a different environment. We continue to work to expand duckplyr's capabilites but would love to hear your experiences trying it out. 
+
+Here are two recordings from last year's posit::conf where we present DuckDB for R and duckplyr:
+
+* [In-Process Analytical Data Management with DuckDB - posit::conf(2023)](https://www.youtube.com/watch?v=9OFzOvV-to4)
+* [duckplyr: Tight Integration of duckdb with R and the tidyverse - posit::conf(2023)](https://www.youtube.com/watch?v=V9GwSPjKMKw)
```
Comment (Mytherin) [inline _posts/2024-04-02-duckplyr.md:229]:
```suggestion
* [duckplyr: Tight Integration of duckdb with R and the tidyverse - posit::conf(2023)](https://www.youtube.com/watch?v=V9GwSPjKMKw)

```

**Review by Mytherin — COMMENTED** [review]
Looks good - some minor comments

**Comment by szarnyasg** [comment]
@hannes I made a few small edits, PTAL at the feedback from Mark, then feel free to hit merge when ready.

**Comment by hannes** [comment]
@szarnyasg I can't apply marks suggestions because of your push I think - can you please incorporate them I'm fine with all suggestions

**Pull request merged**